### PR TITLE
Chore: Allow lambdas with implicit returns

### PR DESCRIPTION
**Pull request by antross**
## Pull request checklist

Make sure you:

- [x] Signed the [Contributor License Agreement](https://cla.js.foundation/webhintio/hint)
- [x] Followed the [commit message guidelines](https://webhint.io/docs/contributor-guide/getting-started/pull-requests/#commit-messages)

For non-trivial changes, please make sure you also:

- ~Added/Updated related documentation.~
- ~Added/Updated related tests.~

## Short description of the change(s)

Enables easier-to-read code in chaining contexts, e.g. when using
`Array` extras such as `filter` and `map`.

### CODE REVIEW

**Comment by molant** [comment]
@antross you should run eslint with the `--fix` to update the code, otherwise this is probably going to fail in the linting phase.

**Comment by molant** [comment]
Now 154 files changed is more what I was expecting @antross 😛

File: packages/connector-jsdom/src/before-parse.ts [hunk 1]
```diff
@@ -11,25 +11,21 @@ import * as jsdomutils from 'jsdom/lib/jsdom/living/generated/utils';
 import readFile from 'hint/dist/src/lib/utils/fs/read-file';
 import { DOMWindow } from 'jsdom';
 
-export const beforeParse = (finalHref: string) => {
-    return (window: DOMWindow) => {
-        const mutationObserverPolyfill = readFile(require.resolve('mutationobserver-shim'));
-        const customElementsPolyfill = readFile(path.join(__dirname, 'polyfills', 'custom-elements.min.js'));
+export const beforeParse = (finalHref: string) => (window: DOMWindow) => {
```
Comment (sarvaje) [inline packages/connector-jsdom/src/before-parse.ts]: IMHO, in this particular case, this was more readable before.

**Comment by sarvaje** [comment]
My heart is divided with this change. In most of the cases is nice to use it, but in some other cases I think we are loosing readability.

**Comment by molant** [comment]
> I think we are loosing readability.

It might take some time for us to get used to it. I think sthis is similar to when we first started using destructuring.

**Comment by antross** [comment]
> I think we are loosing readability.

@sarvaje Have a specific example in mind? We do have a couple of options we can play with.

**Comment by sarvaje** [comment]
Examples:
https://github.com/webhintio/hint/pull/1952/files#diff-67aeed4ecf717994c311d5124e77ebe0R14
https://github.com/webhintio/hint/pull/1952/files#diff-b055eea52f680f438a6a8e04f439718aR39
https://github.com/webhintio/hint/pull/1952/files#diff-493123f8273bb10cbe622ede97d5f2d3R4

I think that in general, when the function has more than one line hehehe.

**Comment by antross** [comment]
I'm with you. Looking at why these are multi-line there is a `requireReturnForObjectLiteral` option that would help in some of these cases. The others seem to need a `requireReturnForFunction` which unfortunately doesn't exist...

**Comment by molant** [comment]
> `requireReturnForFunction`

Time to contribute back?

**Comment by molant** [comment]
This PR is diverging a lot and I don't think we got consensus. Maybe we should close it and give it another try later when we are less busy?

**Pull request closed without merging**